.github/workflows/check-submodule-pointers.yml

```diff
@@ -0,0 +1,25 @@
+name: Check submodule pointers
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  check-submodules:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Compare submodule pointers to lgtm.com branch
+        env: 
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+            if [ $(git rev-parse @:./ql) != $(gh api repos/github/codeql/git/ref/heads/lgtm.com --jq '.object.sha') ] || \
+               [ $(git rev-parse @:./codeql-go) != $(gh api repos/github/codeql-go/git/ref/heads/lgtm.com --jq '.object.sha') ]
```

Comment:
I suggest splitting this, so we can give a clear message for each submodule with the exact commits that were found. Something like:
```bash
CODEQL_ACTUAL_SHA="$(git rev-parse @:./ql)"
CODEQL_EXPECTED_SHA="$(gh api ...)"
if [ "$CODEQL_EXPECTED_SHA" != "$CODEQL_ACTUAL_SHA" ]; then
  echo "... expected $CODEQL_EXPECTED_SHA, found $CODEQL_ACTUAL_SHA ..."
fi
# same for codeql-go
```

One practice I've followed above is: whenever you retrieve the value of a Bash variable, e.g. `$VARNAME`, or a command output, e.g. `$(command)`, it's a good idea to enclose it in double quotes. This handles whitespace in the output value. We don't need it here but it's a good habit.

(We could also use a loop through the pairs of (path, URL), but since there's only two for now it seems like overkill.)